File: nicolaspuebla_proyecto_final/src/main/java/com/example/nicolaspuebla_proyecto_final/model/dataModels/Training.java
package com.example.nicolaspuebla_proyecto_final.model.dataModels;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;

/**
 * Clase que representa la entidad Training, que hereda de Event.
 */
@Entity
@DiscriminatorValue(value = "training")
public class Training extends Event {

    public Training(){}

    public Training(Team team_id, Double latitude, Double longitude, String date) {
        super(team_id, latitude, longitude, date);
    }
}

File: nicolaspuebla_proyecto_final_android/app/src/main/java/com/example/nicolaspuebla_proyecto_final_android/data/model/dataClases/Training.kt
package com.example.nicolaspuebla_proyecto_final_android.data.model.dataClases

/**
 * Clase que almacena la información de un entrenamiento y hereda de Event.
 */
data class Training(
    override val event_type: String,
    override val id: Long,
    override val team: Team,
    override val latitude: Double,
    override val longitude: Double,
    override val date: String,
) : Event(event_type, id, team, latitude, longitude, date)